src/buildercore/cfngen.py

```diff
@@ -623,10 +623,14 @@ def _title_is_removable(title):
     # end backward compatibility code
 
     def _title_has_been_updated(title, section):
-        # title was there before with a deprecated name, leave it alone
-        # e.g. 'EC2Instance' rather than 'EC2Instance1'
-        if not title in old_template[section]:
-            return False
+        if section in old_template:
+            # title was there before with a deprecated name, leave it alone
+            # e.g. 'EC2Instance' rather than 'EC2Instance1'
+            if not title in old_template[section]:
+                return False
+        else:
+            LOG.warn("section %r not present in old template but is present in new: %s" % (section, title))
+            return False # can we handle this better?
```

Comment:
todo: this came up when in `tests.test_cfn.test_altconfig_name_preserved` where the template being generated was missing `Output`s or something. I haven't investigated very much but is probably just an artifact of testing using data in an unexpected way